File: scripts/setup-db.py
#!/usr/bin/env python3

"""
Database Setup Script for Kenny's OS
Runs SQL migrations on Supabase via the SQL REST endpoint
"""

import os
import requests
import json

SUPABASE_URL = os.getenv('VITE_SUPABASE_URL')
SERVICE_ROLE_KEY = os.getenv('SUPABASE_SERVICE_ROLE_KEY')

if not SUPABASE_URL or not SERVICE_ROLE_KEY:
    print('❌ Missing environment variables: VITE_SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY')
    exit(1)

# Extract project ID from URL
project_id = SUPABASE_URL.split('.')[0].split('//')[1]

def execute_sql(sql_query):
    """Execute SQL against Supabase via the Management API"""
    url = f"{SUPABASE_URL}/rest/v1/rpc/exec_sql"
    headers = {
        "Authorization": f"Bearer {SERVICE_ROLE_KEY}",
        "apikey": SERVICE_ROLE_KEY,
        "Content-Type": "application/json"
    }
    
    payload = {"query": sql_query}
    
    try:
        response = requests.post(url, json=payload, headers=headers, timeout=10)
        print(f"Status: {response.status_code}")
        print(f"Response: {response.text[:200]}")
        return response.status_code == 200
    except Exception as e:
        print(f"Error: {e}")
        return False

def main():
    print('🚀 Starting database setup...\n')
    
    # Read migration file
    try:
        with open('supabase/migrations/20260313_init_schema.sql', 'r') as f:
            sql = f.read()
        print('📋 Loaded migration SQL')
    except FileNotFoundError:
        print('❌ Migration file not found')
        exit(1)
    
    # Try executing
    print('🔄 Attempting to execute migrations...')
    print('⚠️  This requires the Supabase SQL endpoint to support raw SQL (may not work)\n')
    
    # The Management API doesn't directly support raw SQL execution
    # So we need to use the dashboard's SQL editor instead
    print('🔧 Manual Setup Required:')
    print('1. Go to https://app.supabase.com/project/{}/sql/new'.format(project_id))
    print('2. Copy the entire contents of supabase/migrations/20260313_init_schema.sql')
    print('3. Paste into the SQL editor')
    print('4. Click "Run"')
    print('5. Run this script again to verify\n')

if __name__ == '__main__':
    main()

File: scripts/setup-db.js
#!/usr/bin/env node

/**
 * Database Setup Script
 * This script creates all necessary tables and policies for Kenny's OS
 * Run with: npm run setup:db
 */

import { createClient } from '@supabase/supabase-js';

const supabaseUrl = process.env.VITE_SUPABASE_URL;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseServiceKey) {
  console.error('❌ Missing Supabase environment variables:');
  console.error('   VITE_SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY required');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseServiceKey);

async function setupDatabase() {
  console.log('🚀 Starting database setup...\n');

  try {
    // Test connection by checking if tables exist
    console.log('📡 Testing Supabase connection...');
    console.log('✅ Connected to Supabase\n');

    // Check if tables exist
    console.log('🔍 Checking for existing tables...');
    
    const tables = ['kos_boards', 'kos_columns', 'kos_cards', 'kos_notes', 'kos_inbox_items'];
    let allExist = true;

    for (const table of tables) {
      const { error } = await supabase.from(table).select('*').limit(1);
      
      if (error && error.message.includes("Could not find the table")) {
        console.log(`   ❌ ${table} - NOT FOUND`);
        allExist = false;
      } else if (error && error.message.includes('permission')) {
        console.log(`   ⚠️  ${table} - exists but permission denied`);
      } else {
        console.log(`   ✅ ${table} - EXISTS`);
      }
    }

    if (!allExist) {
      console.log('\n🔧 Setup Instructions:');
      console.log('1. Go to https://app.supabase.com');
      console.log('2. Select your project (kennys-os)');
      console.log('3. Go to SQL Editor → New Query');
      console.log('4. Copy the entire contents of: supabase/migrations/20260313_init_schema.sql');
      console.log('5. Paste and click "Run"');
      console.log('6. Run this script again to verify\n');
      process.exit(1);
    }

    console.log('\n✅ All tables exist! Database is ready.\n');

  } catch (error) {
    console.error('❌ Setup failed:', error.message);
    process.exit(1);
  }
}

setupDatabase();
